.github/workflows/scripts/test-container.rb

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# Based on Sam Ruby's system test from dockerfile-rails:
+# https://github.com/rubys/dockerfile-rails/pull/21
+
+require "net/http"
+
+LOCALHOST = `uname -n`.chomp
+PORT = 3000
+
+60.times do |i|
+  sleep 0.5
+  begin
+    response = Net::HTTP.get_response(LOCALHOST, "/up", PORT)
+
+    if %w(404 500).include? response.code
+      status = response.code.to_i
+    end
+
+    puts response.body
+    exit status || 0
+  rescue Errno::ECONNREFUSED, Errno::ETIMEDOUT, EOFError
```

Comment:
```suggestion
  rescue SystemCallError, IOError
```

Not 100% sure about that one, but Net::HTTP leak tons of different errors, so we might as well have a wide rescue, see: https://github.com/lostisland/faraday-net_http/blob/16ad984dfc0e2915addda17aa6879d8158ce32ad/lib/faraday/adapter/net_http.rb#L15-L30